pkgs/development/compilers/temurin-bin/jdk-darwin-base.nix

```diff
@@ -0,0 +1,67 @@
+{ name-prefix ? "temurin"
+, brand-name ? "Eclipse Temurin"
+, sourcePerArch
+, knownVulnerabilities ? []
+}:
+
+{ swingSupport ? true # not used for now
+, lib, stdenv
+, fetchurl
+, setJavaClassPath
+}:
+
+let cpuName = stdenv.hostPlatform.parsed.cpu.name;
+    result = stdenv.mkDerivation {
+  name = if sourcePerArch.packageType == "jdk"
+    then "${name-prefix}-bin-${sourcePerArch.${cpuName}.version}"
+    else "${name-prefix}-${sourcePerArch.packageType}-bin-${sourcePerArch.${cpuName}.version}";
+
+  src = fetchurl {
+    inherit (sourcePerArch.${cpuName}) url sha256;
+  };
+
+  # See: https://github.com/NixOS/patchelf/issues/10
+  dontStrip = 1;
+
+  installPhase = ''
+    cd ..
+
+    mv $sourceRoot $out
+
+    # jni.h expects jni_md.h to be in the header search path.
+    ln -s $out/Contents/Home/include/darwin/*_md.h $out/Contents/Home/include/
+
+    # Remove some broken manpages.
+    # Only for 11 and earlier.
+    [ -e "$out/Contents/Home/man/ja" ] && rm -r $out/Contents/Home/man/ja
+
+    ln -s $out/Contents/Home/* $out/
+
+    # Propagate the setJavaClassPath setup hook from the JDK so that
+    # any package that depends on the JDK has $CLASSPATH set up
+    # properly.
+    mkdir -p $out/nix-support
+    printWords ${setJavaClassPath} > $out/nix-support/propagated-build-inputs
+
+    # Set JAVA_HOME automatically.
+    cat <<EOF >> $out/nix-support/setup-hook
+    if [ -z "\''${JAVA_HOME-}" ]; then export JAVA_HOME=$out; fi
+    EOF
+  '';
+
+  # FIXME: use multiple outputs or return actual JRE package
+  passthru = {
+    jre = result;
+    home = result;
+  };
+
+  meta = with lib; {
+    license = licenses.gpl2Classpath;
+    description = "${brand-name}, prebuilt OpenJDK binary";
+    platforms = [ "x86_64-darwin" ]; # some inherit jre.meta.platforms
```

Comment:
On an aarch64 Mac this causes unsupported-platform errors for me:
```
error: Package ‘temurin-bin-17.0.1’ in /nix/store/2amg850cxgs627bix28nm0m5zq4fv38k-source/pkgs/development/compilers/temurin-bin/jdk-darwin-base.nix:60 is not supported on ‘aarch64-darwin’, refusing to evaluate.
```

I tested around a bit and came to this solution: https://github.com/taku0/nixpkgs/compare/temurin-17...pmenke-de:pmenke-de_temurin-17

Maybe some of these adaptions could be used in this PR.
I'm not 100% sure on the `meta.platforms` end of things though, as I don't know how this works, when the same-named package is built from different sources on different platforms. Should `meta.platforms` list all theoretically supported platforms in this case, or should it only list the specific platform(s), this specific instance of the package runs on?

Reply:
We should able to filter packages by platform without building it.  Since `version` or other attributes are evaluated lazily, the error is not thrown until the build time. Therefore, we should list actually supported platforms in `platforms` rather than relying on errors on the build time.

Because I don't have M1 Macs, I'm hesitating to add `aarch64-darwin` to `platforms`.

Reply:
> We should able to filter packages by platform without building it. Since version or other attributes are evaluated lazily, the error is not thrown until the build time. Therefore, we should list actually supported platforms in platforms rather than relying on errors on the build time.

Ok. That's good to know. Thank you.

> Because I don't have M1 Macs, I'm hesitating to add `aarch64-darwin` to `platforms`.

Understandable. I do have a M1 Mac though and thus would appreciate, if we could include `aarch64-darwin` as supported platform. If I could somehow help this effort - e.g. by testing or sth. please let me know.
So far I can at least say, that it builds and runs on my machine :-)

In my above mentioned commit (https://github.com/taku0/nixpkgs/compare/temurin-17...pmenke-de:pmenke-de_temurin-17) - that I changed/updated in the meantime - I also addressed one more thing:

This line in [jdk-linux-base.nix](https://github.com/NixOS/nixpkgs/pull/140364/files#diff-8b64e051aa30db104a76317d4ca010753c1415eb83d5b1e75865b0ba1ac8def7R119) generates some unexpected platform entries, from non-platform like keys in the sources.json:
```nix
platforms = lib.mapAttrsToList (arch: _: arch + "-linux") sourcePerArch
```

```nix
nix-repl> outputs.legacyPackages.x86_64-linux.temurin-bin-11.meta.platforms
[ "aarch64-linux" "armv6l-linux" "armv7l-linux" "packageType-linux" "vmType-linux" "x86_64-linux" ]
```

Maybe sth. like this would be better? - just looking at the expression output; I'm not experienced enough to tell whether this is a "well-written" nix-expression.
```nix
platforms = builtins.map (arch: arch + "-linux") (builtins.filter ( arch: builtins.elem arch (builtins.attrNames lib.systems.parse.cpuTypes) ) (builtins.attrNames sourcePerArch));
```
```nix
nix-repl> outputs.legacyPackages.x86_64-linux.temurin-bin-11.meta.platforms
[ "aarch64-linux" "armv6l-linux" "armv7l-linux" "x86_64-linux" ]
```

Reply:
@pmenke-de I have incorporated your idea. Thank you. Could you try it on M1 Mac?

Reply:
@taku0 I tested it on my machine. Works perfectly 👍 
Thanks for implementing it.